zero/src/prover.rs

```diff
@@ -131,29 +135,180 @@ impl BlockProverInput {
+                    let mut segment_counter = 0;
+                    let mut segment_proving_tasks = Vec::new();
+                    while let Some(segment_data) = segment_rx.recv().await {
+                        if segment_data.is_none() {
+                            break;
+                        }
+                        let seg_prove_ops = seg_prove_ops.clone();
+                        let proof_runtime = proof_runtime.clone();
+                        let segment_proof_tx = segment_proof_tx.clone();
+                        // Prove one segment in a dedicated async task.
+                        let segment_proving_task = tokio::spawn(async move {
+                            if let Some(segment_data) = segment_data {
+                                debug!("proving the batch {batch_idx} segment nr. {segment_counter}");
+                                let seg_aggregatable_proof= Directive::map(
+                                    IndexedStream::from([segment_data]),
+                                    &seg_prove_ops,
+                                )
+                                    .run(&proof_runtime.heavy_proof)
+                                    .await?
+                                    .into_values_sorted()
+                                    .await?
+                                    .into_iter()
+                                    .next()
+                                    .context(format!(
+                                        "failed to get segment proof, batch: {batch_idx}, segment: {segment_counter}"
+                                    ))?;
+
+                                segment_proof_tx
+                                    .send((segment_counter, seg_aggregatable_proof))
+                                    .await
+                                    .context(format!(
+                                        "unable to send segment proof, batch: {batch_idx}, segment: {segment_counter}"
+                                    ))?;
+                            };
+                            Ok::<(), anyhow::Error>(())
+                        });
+
+                        segment_proving_tasks.push(segment_proving_task);
+                        segment_counter += 1;
+                    }
+                    drop(segment_proof_tx);
+                    // Wait for all the segment proving tasks of one batch to finish.
+                    while let Some((segment_idx, segment_aggregatable_proof)) = segment_proof_rx.recv().await {
+                        batch_segment_aggregatable_proofs.push((segment_idx, segment_aggregatable_proof));
+                    }
+                    try_join_all(segment_proving_tasks).await?;
+                    batch_segment_aggregatable_proofs.sort_by(|(a, _), (b, _)| a.cmp(b));
+                    debug!(block_number=%block_number, batch=%batch_idx, "finished proving all segments");
```

Comment:
nit: [logging](https://0xpolygon.slack.com/archives/C073HA1NSDU/p1727888673088639)
```suggestion
                    debug!(%block_number, batch=%batch_idx, "finished proving all segments");
```